paddle/fluid/operators/distributed_ops/gen_nccl_id_op.cc

```diff
@@ -118,23 +250,31 @@ GenNCCLId operator
 For trainer 0: generate a new UniqueId and send it to all the other trainers.
 For trainer 1~n: start a gRPC server to get the UniqueId, once got, stop the server.
 )DOC");
-    AddAttr<std::string>("endpoint",
-                         "(string), e.g. 127.0.0.1:6175 "
-                         "current listen endpoint");
     AddAttr<std::vector<std::string>>(
-        "endpoint_list",
-        "['trainer1_ip:port', 'trainer2_ip:port', ...] "
-        "list of trainer endpoints start from trainer 1")
+        "trainers",
+        "['trainer0_ip:port', 'trainer1_ip:port', ...] "
+        "list of all trainer endpoints")
         .SetDefault({});
     AddAttr<int>("trainer_id",
-                 "(int default 0) "
-                 "The index of the trainer in distributed training.")
-        .SetDefault(0);
+                 "(int) "
+                 "The index of the trainer in distributed training.");
+    AddAttr<int>("nccl_comm_num",
+                 "(int default 1) "
+                 "The number of nccl communicator num.")
+        .SetDefault(1);
+    AddAttr<bool>("use_hierarchical_allreduce",
+                  "(bool default false) "
+                  "Wheter to use hierarchical allreduce.")
+        .SetDefault(false);
+    AddAttr<int>("hierarchical_allreduce_inter_nranks",
+                 "(int default 1) "
+                 "Wheter to use hierarchical allreduce.")
+        .SetDefault(-1);
```

Comment:
Considering compatibility issues, you should add the default value for those added attributions.

Reply:
There should be an invalid value to check it later when running.